src/vconsole/vconsole-setup.c

```diff
@@ -516,19 +516,26 @@ static void setup_remaining_vcs(int src_fd, unsigned src_idx, bool utf8) {
 }
 
 static int find_source_vc(char **ret_path, unsigned *ret_idx) {
-        int r, err = 0;
+        int err = 0;
 
         assert(ret_path);
         assert(ret_idx);
 
+        /* This function returns an fd when it finds a candidate. It returns 0 when it finds some VCs but all
+         * are busy. Otherwise it returns the first error that occured when the VC was opened. */
```

Comment:
0 is a valid fd. don't use that as a marker for a special case please.

just change the function prototype to return the fd it returns via return value via a return param instead, i.e. add `int **ret_fd` to the param list. that frees the ret val for you to communicate whether we found a VC or not.

Reply:
I'll fix it.